hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractWALRoller.java

```diff
@@ -232,7 +230,7 @@ private boolean isWaiting() {
    * @return true if all WAL roll finished
    */
   public boolean walRollFinished() {
-    return walNeedsRoll.values().stream().allMatch(needRoll -> !needRoll) && isWaiting();
```

Comment:
Checking its usage, I think this API impl is already buggy.   This just checks the status of the boolean. Once we start a roll on a WAL, we reset the boolean (Even before this patch).  So it is not clearly telling anything abt the roll status. This can return true even while an active wal roll is going on.  We can keep it as an another jira and fix (if required).. Just add some TODO comments here.
We might need another boolean in Controller which clearly tracks whether we are ongoing a roll. So this really need to check that status as well as a requested roll status.